.github/workflows/integration_test.yml

```diff
@@ -14,4 +14,4 @@ jobs:
 
       - name: Run integration tests
         run: |
-          ./aws/logs_monitoring/tools/integration_tests/integration_tests.sh --python-version=${{ matrix.python-version }}
+          ./aws/logs_monitoring/tools/integration_tests/integration_tests.sh --python-version=${{ matrix.python-version }} --no-additional-lambda
```

Comment:
Why `--no-additional-lambda`? What's the concerns?

Reply:
For the additional lambda case we need to provision a couple of lambdas and get their logs from cloudwatch. So locally you would run this using `aws-vault` since these run in Github I'm not sure if we have a secure way to set this up.

Reply:
Got it, perhaps we could drop the requirements here to just have a unit test? Given the nature that AWS is responsible for invoking the lambda function, integration test might be an overkill and we just to ensure `invoke` is indeed called?

My concern is that after a while the integration tests for this case will be broken and unmaintained since it does not run automatically.

Reply:
that makes sense. I'll add a unit test instead and delete that integration test changes

Reply:
Alternatively, I could make the default to run without those lambda cases. And make sure that an option is required to actually run with the lambda case.

Reply:
curious to get @DarcyRaynerDD thoughts here too.

Reply:
Yeah, definitely does not hurt to keep the code if you like, assuming you change the default to not run them.